plugins/org.yakindu.sct.generator.c/src/org/yakindu/sct/generator/c/HeaderContentFragmentProvider.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2018 committers of YAKINDU and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * Contributors:
+ * 	rbeckmann - initial API and implementation
+ *
+ */
+package org.yakindu.sct.generator.c;
+
+import static org.yakindu.sct.model.stext.lib.StatechartAnnotations.EVENT_DRIVEN_ANNOTATION;
+
+import org.yakindu.base.types.Annotation;
+import org.yakindu.sct.generator.c.files.StatemachineHeader;
+import org.yakindu.sct.generator.c.submodules.StatemachineHeaderContentFragment;
+import org.yakindu.sct.generator.c.submodules.eventdriven.EventDrivenStatemachineHeaderFragment;
+import org.yakindu.sct.model.sgen.GeneratorEntry;
+import org.yakindu.sct.model.sgraph.Statechart;
+
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Provider;
+
+/**
+ * @author rbeckmann
+ *
+ */
+public class HeaderContentFragmentProvider implements Provider<StatemachineHeader> {
+	@Inject
+	Injector injector;
+
+	@Override
+	public StatemachineHeader get() {
+		StatemachineHeader source = new StatemachineHeader();
```

Comment:
why not injector.getInstance(StatemachineHeader.class) here?
if 'new' is used here binding overriding will not work!

Reply:
I can tell you why - it leads to StackOverflowErrors to get an instance via injector in the provider that is bound in the module. ;) This provider should always provide StatemachineHeader, nothing else, and you would rebind it to another provider.